src/ext/png.c

```diff
@@ -141,14 +165,25 @@ png_buffer png_write(png_img src)
         PNG_FILTER_TYPE_DEFAULT
     );
 
+    // Save cartridge data in a chunk too. This supports bigger cartridges than steganography
+    if (cart.data && cart.size > 0 && cart.size <= 0x7fffff){
```

Comment:
Here you store the cart data in the `caRt` chunk in addition to the steganographic data and it looks redundant, we are doubling the cart data in two places. I suggest letting the user decide how to save the `.png` by adding console command options like `save cart.png [way=(steno | caRt | both)]`.
What do you think?

Reply:
(bztsrc is having issues logging into github and asked someone to forward this response.)

    > Here you store the cart data in the caRt chunk in addition to the steganographic data and it looks redundant, we are doubling the cart data in two places.

    Only for now to provide backward compatibility.

    > I suggest letting the user decide

    I think it would be much wiser to abandon steganography entirely instead. My reasons:
    - steganography has very limited storage capacity (98k tops)
    - it silently fails if the cartridge data happens to be bigger than that
    - it is extremely vulnerable to data corruption
    - has no means at all to detect data corruption

    On the other hand, chunk based storage
    - has unlimited storage capacity (up to 2^31)
    - can't silently fail on save, it always works
    - has a CRC to detect data corruption (handled by libpng automatically)

Reply:
I agree.  I think we need to maintain backwards compatibility - to continue to READ old steganography but if you make changes and save that cartridge it should be encoded as PNG chunk metadata.  Giving the user 3 different choices is kind of the word of all approaches.